java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        label = ((ContinueStatementTree) s).label().name();
+      }
+      Block target = labels.get(label);
+      if (target == null) {
+        throw new IllegalStateException("Undeclared label: " + label);
+      }
+      b.successors.add(target);
+    }
+
+    for (Block b : blocks) {
+      for (Block successor : b.successors) {
+        successor.predecessors.add(b);
+      }
+    }
+  }
+
+  private Block createBlock(Block successor) {
+    Block result = createBlock();
+    result.successors.add(successor);
+    return result;
+  }
+
+  private Block createBlock() {
+    Block result = new Block(blocks.size());
+    blocks.add(result);
+    return result;
+  }
+
+  public static CFG build(MethodTree tree) {
+    Preconditions.checkArgument(tree.block() != null, "Cannot build CFG for method with no body.");
+    return new CFG(tree.block());
+  }
+
+  private void build(List<? extends Tree> trees) {
+    for (Tree tree : Lists.reverse(trees)) {
+      build(tree);
+    }
+  }
+
+  private void build(Tree tree) {
+    switch (tree.kind()) {
+      case BLOCK:
+        for (StatementTree statementTree : Lists.reverse(((BlockTree) tree).body())) {
+          build(statementTree);
+        }
+        break;
+      case RETURN_STATEMENT: {
+        ReturnStatementTree s = (ReturnStatementTree) tree;
+        currentBlock = createUnconditionalJump(s, exitBlock);
+        ExpressionTree expression = s.expression();
+        if (expression != null) {
+          build(expression);
+        }
+        break;
+      }
+      case EXPRESSION_STATEMENT:
+        build(((ExpressionStatementTree) tree).expression());
+        break;
+      case METHOD_INVOCATION:
+        MethodInvocationTree mit = (MethodInvocationTree) tree;
+        currentBlock.elements.add(mit);
+        build(mit.methodSelect());
+        for (ExpressionTree arg : Lists.reverse(mit.arguments())) {
+          build(arg);
+        }
+        break;
+      case IF_STATEMENT: {
+        IfStatementTree ifStatementTree = (IfStatementTree) tree;
+        Block next = currentBlock;
+        // process else-branch
+        Block elseBlock = next;
+        StatementTree elseStatement = ifStatementTree.elseStatement();
+        if (elseStatement != null) {
+          // if statement will create the required block.
+          if (!elseStatement.is(Tree.Kind.IF_STATEMENT)) {
+            currentBlock = createBlock(next);
+          }
+          build(elseStatement);
+          elseBlock = currentBlock;
+        }
+        // process then-branch
+        currentBlock = createBlock(next);
+        build(ifStatementTree.thenStatement());
+        Block thenBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(ifStatementTree, thenBlock, elseBlock);
+        buildCondition(ifStatementTree.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case CONDITIONAL_EXPRESSION: {
+        ConditionalExpressionTree cond = (ConditionalExpressionTree) tree;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 11. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)